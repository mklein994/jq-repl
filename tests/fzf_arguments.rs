use std::process::{Command, Stdio};

#[test]
fn check_fzf_command_output() {
    let output = Command::new(env!(concat!("CARGO_BIN_EXE_", clap::crate_name!())))
        .env_remove("EDITOR")
        .env_remove("PAGER")
        .arg("--history-file")
        .arg("/tmp/jq_repl_history")
        .arg("--show-fzf-command")
        .arg("-n")
        .stdin(Stdio::inherit())
        .output()
        .unwrap();

    let stdout = String::from_utf8(output.stdout).unwrap();

    let line = line!();
    let expected = r##"#!/bin/bash

fzf \
--disabled \
$'--preview-window=up,99%,border-bottom' \
$'--info=hidden' \
--header-first \
$'--prompt=-n> ' \
$'--header=M-e: editor \xE2\x81\x84 M-v: vd \xE2\x81\x84 M-l: pager \xE2\x81\x84 ^<space>: gron' \
$'--history=/tmp/jq_repl_history' \
$'--preview=gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ' \
$'--bind=ctrl-k:kill-line,pgup:preview-page-up,pgdn:preview-page-down,alt-w:toggle-preview-wrap,home:preview-top,end:preview-bottom' \
$'--bind=alt-s:change-prompt(-ns> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n --slurp {q} ' \
$'--bind=alt-S:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ' \
$'--bind=alt-c:change-prompt(-nc> )+preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -c {q} ' \
$'--bind=alt-C:change-prompt(-n> )+preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ' \
$'--bind=ctrl-space:change-prompt(-n gron> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q}  | gron --colorize' \
$'--bind=alt-space:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ' \
$'--bind=alt-e:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q}  | nvim -c \'set ft=json\' -' \
$'--bind=alt-v:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q}  | vd --filetype json' \
$'--bind=alt-V:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q}  | vd --filetype csv' \
$'--bind=alt-l:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q}  | less -R' \
$'--bind=alt-L:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q}  | bat --language json' < /dev/null
"##;

    assert_eq!(
        expected.lines().count(),
        stdout.lines().count(),
        "expected != actual line count.\nstdout:\n{stdout}\n\nstderr:\n{}",
        String::from_utf8_lossy(&output.stderr)
    );

    if expected != stdout {
        for (line_number, (expected_line, actual_line)) in
            expected.lines().zip(stdout.lines()).enumerate()
        {
            assert_eq!(
                expected_line,
                actual_line,
                "failed on line {}",
                line as usize + line_number + 1
            );
        }
    }
}
