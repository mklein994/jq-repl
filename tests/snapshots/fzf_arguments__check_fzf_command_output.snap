---
source: tests/fzf_arguments.rs
expression: stdout
---
#!/bin/bash

fzf \
--disabled \
$'--preview-window=up,99%,border-bottom' \
--no-separator \
$'--info=hidden' \
--header-first \
$'--query=.' \
$'--prompt=-n> ' \
$'--header=M-e: editor \xE2\x81\x84 M-j: vd \xE2\x81\x84 M-l: pager \xE2\x81\x84 M-g: braille \xE2\x81\x84 ^<space>: gron' \
$'--history=/tmp/jq_repl_history' \
$'--preview-label-pos=-1' \
$'--bind=change:transform-preview-label:printf "%s" {q} | charcounter ' \
$'--preview=gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ./tests/foo.json' \
$'--bind=ctrl-k:kill-line,pgup:preview-page-up,pgdn:preview-page-down,alt-w:toggle-preview-wrap,home:preview-top,end:preview-bottom,tab:refresh-preview' \
$'--bind=alt-s:change-prompt(-ns> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n --slurp {q} ./tests/foo.json' \
$'--bind=alt-S:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ./tests/foo.json' \
$'--bind=alt-c:change-prompt(-nc> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -c {q} ./tests/foo.json' \
$'--bind=alt-C:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ./tests/foo.json' \
$'--bind=ctrl-space:change-prompt(-n gron> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | gron --colorize' \
$'--bind=alt-space:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ./tests/foo.json' \
$'--bind=alt-g:change-prompt(-n braille> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | BRAILLE_USE_FULL_DEFAULT_HEIGHT=1 braille --modeline' \
$'--bind=alt-G:change-prompt(-n> )+change-preview:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n {q} ./tests/foo.json' \
$'--bind=alt-e:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | nvim -c \'set ft=json\' -' \
$'--bind=alt-j:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | vd --filetype json' \
$'--bind=alt-J:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -c -M {q} ./tests/foo.json | vd --filetype jsonl' \
$'--bind=alt-v:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | vd --filetype csv' \
$'--bind=alt-l:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -c -M {q} ./tests/foo.json | less ' \
$'--bind=alt-L:execute:gojq -L ~/.jq -L ~/.jq/.jq --raw-output -C -n -M {q} ./tests/foo.json | bat --language json --paging always' < /dev/null
